diff -Naurp pcp-3.10.3.orig/src/pmdas/postfix/Install pcp-3.10.3/src/pmdas/postfix/Install
--- pcp-3.10.3.orig/src/pmdas/postfix/Install	2015-01-21 02:18:01.000000000 +1100
+++ pcp-3.10.3/src/pmdas/postfix/Install	2015-05-04 17:26:02.021177109 +1000
@@ -21,7 +21,7 @@
 iam=postfix
 perl_opt=true
 daemon_opt=false
-forced_restart=false
+forced_restart=true
 
 which qshape >/dev/null 2>&1
 if test $? -ne 0; then
diff -Naurp pcp-3.10.3.orig/src/pmdas/postfix/pmdapostfix.pl pcp-3.10.3/src/pmdas/postfix/pmdapostfix.pl
--- pcp-3.10.3.orig/src/pmdas/postfix/pmdapostfix.pl	2015-01-21 02:18:01.000000000 +1100
+++ pcp-3.10.3/src/pmdas/postfix/pmdapostfix.pl	2015-05-04 17:26:06.825180800 +1000
@@ -211,5 +211,4 @@ $pmda->add_indom($postfix_sent_indom, \@
 $pmda->add_indom($postfix_received_indom, \@postfix_received_dom, '', '');
 $pmda->add_tail($logfile, \&postfix_log_parser, 0);
 $pmda->set_fetch_callback(\&postfix_fetch_callback);
-$pmda->set_user('postfix');
 $pmda->run;
