commit dac8621e4d9b70cf5891ba00307dc1c4fdf9013a
Author: Nathan Scott <nathans@redhat.com>
Date:   Wed Oct 25 11:32:46 2017 +1100

    build: multilib strikes again - updates since autoreconf work

diff --git a/src/include/pcp/config.h.in b/src/include/pcp/config.h.in
index f9376b8b3..72b4e825e 100644
--- a/src/include/pcp/config.h.in
+++ b/src/include/pcp/config.h.in
@@ -3,18 +3,6 @@
 /* Define to 1 if `TIOCGWINSZ' requires <sys/ioctl.h>. */
 #undef GWINSZ_IN_SYS_IOCTL
 
-/* sizeof long */
-#undef HAVE_32BIT_LONG
-
-/* pointer size */
-#undef HAVE_32BIT_PTR
-
-/* sizeof long */
-#undef HAVE_64BIT_LONG
-
-/* pointer size */
-#undef HAVE_64BIT_PTR
-
 /* AI_ADDRCONFIG macro */
 #undef HAVE_AI_ADDRCONFIG
 
@@ -48,6 +36,9 @@
 /* Define to 1 if you have the <bits/wordsize.h> header file. */
 #undef HAVE_BITS_WORDSIZE_H
 
+/* Define to 1 if you have the <boost/flyweight.hpp> header file. */
+#undef HAVE_BOOST_FLYWEIGHT_HPP
+
 /* Define to 1 if you have the `brk' function. */
 #undef HAVE_BRK
 
@@ -697,12 +688,6 @@
 /* Define to the version of this package. */
 #undef PACKAGE_VERSION
 
-/* sizeof suseconds_t */
-#undef PM_SIZEOF_SUSECONDS_T
-
-/* sizeof time_t */
-#undef PM_SIZEOF_TIME_T
-
 /* Define as the return type of signal handlers (`int' or `void'). */
 #undef RETSIGTYPE
 
diff --git a/src/include/pcp/configsz.h.in b/src/include/pcp/configsz.h.in
index d983b6e38..18d35197a 100644
--- a/src/include/pcp/configsz.h.in
+++ b/src/include/pcp/configsz.h.in
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2014,2016 Red Hat.
+ * Copyright (c) 2014-2017 Red Hat.
  *
  * This library is free software; you can redistribute it and/or modify it
  * under the terms of the GNU Lesser General Public License as published
@@ -14,20 +14,28 @@
 #ifndef PCP_CONFIGSZ_H
 #define PCP_CONFIGSZ_H
 
-/* long and pointer must be either 32 bit or 64 bit */
+/* sizeof long */
 #undef HAVE_64BIT_LONG
+
+/* sizeof long */
 #undef HAVE_32BIT_LONG
+
+/* pointer size */
 #undef HAVE_32BIT_PTR
+
+/* pointer size */
 #undef HAVE_64BIT_PTR
 
-/* Check size of suseconds_t type (struct timeval) */
+/* sizeof suseconds_t */
 #undef PM_SIZEOF_SUSECONDS_T
+
 #ifndef PM_SIZEOF_SUSECONDS_T
 #error Unknown suseconds_t size
 #endif
 
-/* Check size of time_t (struct timeval, timespec) */
+/* sizeof time_t */
 #undef PM_SIZEOF_TIME_T
+
 #ifndef PM_SIZEOF_TIME_T
 #error Unknown time_t size
 #endif
